tests/vmi_configuration_test.go

```diff
@@ -1959,11 +1960,13 @@ var _ = Describe("Configurations", func() {
 				Expect(err).ToNot(HaveOccurred())
 				defer expecter.Close()
 
-				res, err := expecter.ExpectBatch([]expect.Batcher{
-					&expect.BSnd{S: "[ $(free -m | grep Mem: | tr -s ' ' | cut -d' ' -f2) -lt 80 ] && echo 'pass' || echo 'fail'\n"},
-					&expect.BExp{R: "pass"},
-					&expect.BSnd{S: "swapoff -a && dd if=/dev/zero of=/dev/shm/test bs=1k count=118k; echo $?\n"},
-					&expect.BExp{R: "0"},
+				res, err := tests.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
+					&expect.BSnd{S: "[ $(free -m | grep Mem: | tr -s ' ' | cut -d' ' -f2) -lt 80 ] && echo 'pass' || echo 'fail'\r"},
+					&expect.BExp{R: "\npass.*\\$"},
```

Comment:
can tests.Retcode be used ? or because the .* it cant ?

Reply:
Yes, I think it can be used. The purpose of the the retcode was to use it for `echo $?`. But you're right, it is a good idea to use it for any value we search for.
Once this PR is merged we should introduce followup PRs using the Retcode for expected values and  tests.ExpectBatchWithValidatedSend instead of `expect.ExpectBatch`.

We should have constants for the prompts or specifying the same prompt for all the vm types on login.